states/statefile/version4.go

```diff
@@ -452,13 +471,24 @@ func appendInstanceObjectStateV4(rs *states.Resource, is *states.ResourceInstanc
 		}
 	}
 
+	// Extract paths from path value marks
+	var paths []cty.Path
+	for _, vm := range obj.AttrSensitivePaths {
+		paths = append(paths, vm.Path)
+	}
+
+	// Marshal paths to JSON
+	attributePaths, pathsDiags := marshalPaths(paths)
+	diags = diags.Append(pathsDiags)
+
 	return append(isV4s, instanceObjectStateV4{
 		IndexKey:            rawKey,
 		Deposed:             string(deposed),
 		Status:              status,
 		SchemaVersion:       obj.SchemaVersion,
 		AttributesFlat:      obj.AttrsFlat,
 		AttributesRaw:       obj.AttrsJSON,
+		AttributePaths:      attributePaths,
```

Comment:
I see that some of the other things around this are now renamed to "sensitive paths" or similar; perhaps it would make sense to rename this one too? This doesn't have the same "would need a future breaking change" weight as some of the other ones, but the inconsistency tripped me up a bit here when I was reading through.